src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,404 @@
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForLengthPattern, "[0]").WithArguments("int").WithLocation(7, 18),
+                // (8,18): error CS9202: Length patterns may not be used for a value of type 'int'.
+                //         _ = 0 is [0]{0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForLengthPattern, "[0]").WithArguments("int").WithLocation(8, 18),
+                // (9,18): error CS9200: List patterns may not be used for a value of type 'int'.
+                //         _ = 0 is {..0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{..0}").WithArguments("int").WithLocation(9, 18),
+                // (9,19): error CS9201: Slice patterns may not be used for a value of type 'int'.
+                //         _ = 0 is {..0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForSlicePattern, "..0").WithArguments("int").WithLocation(9, 19));
+        }
+
+        [Fact]
+        public void ListPattern_ObsoleteMembers()
+        {
+            var source = @"
+using System;
+class X
+{
+    [Obsolete(""error"", error: true)]
+    public int Slice(int i, int j) => 0;
+    [Obsolete(""error"", error: true)]
+    public int this[int i] => 0;
+    [Obsolete(""error"", error: true)]
+    public int Count => 0;
+
+    public void M()
+    {
+        _ = this is {0};
+        _ = this is [0];
+        _ = this is [1]{0};
+        _ = this is {..0};
+    } 
+}
+";
+            // PROTOTYPE(list-patterns) Missing error for [0]: https://github.com/dotnet/roslyn/issues/53418
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithListPatterns);
+            compilation.VerifyDiagnostics(
+                // (14,21): error CS0619: 'X.this[int]' is obsolete: 'error'
+                //         _ = this is {0};
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "{0}").WithArguments("X.this[int]", "error").WithLocation(14, 21),
+                // (16,24): error CS0619: 'X.this[int]' is obsolete: 'error'
+                //         _ = this is [1]{0};
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "{0}").WithArguments("X.this[int]", "error").WithLocation(16, 24),
+                // (17,21): error CS0619: 'X.this[int]' is obsolete: 'error'
+                //         _ = this is {..0};
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "{..0}").WithArguments("X.this[int]", "error").WithLocation(17, 21));
+        }
+
+        [Fact]
+        public void SlicePattern_Misplaced()
+        {
+            var source = @"
+class X
+{
+    public void M(int[] a)
+    {
+        _ = a is {.., ..};
+        _ = a is {(..)};
+    } 
+}
+";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithListPatterns);
+            compilation.VerifyDiagnostics(
+                // (6,23): error CS9203: Slice patterns may only be used once and directly inside a list pattern.
```

Comment:
The error message doesn't describe the problem (missing pattern inside slice pattern). Can we tweak the message?

Reply:
The error message is correct. the pattern following `..` is optional.

Reply:
Got it. Thanks